src/Analysis/Engine/Impl/Values/Protocols.cs

```diff
@@ -389,9 +390,41 @@ class TupleProtocol : IterableProtocol {
 
         public TupleProtocol(ProtocolInfo self, IEnumerable<IAnalysisSet> values) : base(self, AnalysisSet.UnionAll(values)) {
             _values = values.Select(s => s.AsUnion(1)).ToArray();
+            Name = GetNameFromValues();
+        }
 
-            var argTypes = _values.SelectMany(e => e.Select(v => v is IHasQualifiedName qn ? qn.FullyQualifiedName : v.ShortDescription));
-            Name = "tuple[{0}]".FormatInvariant(string.Join(", ", argTypes));
+        private string GetNameFromValues() {
+            // Enumerate manually since SelectMany drops empty/unknown values
+            var sb = new StringBuilder("tuple[");
+            for (var i = 0; i < _values.Length; i++) {
+                if (i > 0) {
```

Comment:
We can have an extension method for this:
https://github.com/Microsoft/RTVS/blob/master/src/Common/Core/Impl/Extensions/StringBuilderExtensions.cs